src/NuGet.Core/NuGet.Commands/RestoreCommand.cs

```diff
@@ -723,13 +746,11 @@ private bool ResolutionSucceeded(IEnumerable<RestoreTargetGraph> graphs)
                             Framework = projectFramework,
 
                             // Find all dependencies which would be in the nuspec
-                            // If there is no
+                            // Include dependencies with no constraints, or package/project/external
                             Dependencies = graphItem.Data.Dependencies
                                 .Where(
-                                    d => d.LibraryRange.TypeConstraint == null
-                                    || d.LibraryRange.TypeConstraint == LibraryTypes.Package
-                                    || d.LibraryRange.TypeConstraint == LibraryTypes.Project
-                                    || d.LibraryRange.TypeConstraint == LibraryTypes.ExternalProject)
+                                    d => (d.LibraryRange.TypeConstraint & LibraryTargetFlagUtils.PackageProjectExternal) 
+                                                != LibraryTypeFlag.None)
```

Comment:
Extension method? `d.LibraryRange.TypeConstraint.Is[blahblah]` would be a little clearer IMO